contracts/contracts/strategies/aerodrome/AerodromeAMOStrategy.sol

```diff
@@ -0,0 +1,964 @@
+     * @dev Un-stakes the token from the gauge for the execution duration of
+     * the function and after that re-stakes it back in.
+     *
+     * It is important that the token is unstaked and owned by the strategy contract
+     * during any liquidity altering operations and that it is re-staked back into the
+     * gauge after liquidity changes. If the token fails to re-stake back to the
+     * gauge it is not earning incentives.
+     */
+    // all functions using this modifier are used by functions with reentrancy check
+    // slither-disable-start reentrancy-no-eth
+    modifier gaugeUnstakeAndRestake() {
+        // because of solidity short-circuit _isLpTokenStakedToGauge doesn't get called
+        // when tokenId == 0
+        if (tokenId != 0 && _isLpTokenStakedToGauge()) {
+            clGauge.withdraw(tokenId);
+        }
+        _;
+        // because of solidity short-circuit _isLpTokenStakedToGauge doesn't get called
+        // when tokenId == 0
+        if (tokenId != 0 && !_isLpTokenStakedToGauge()) {
+            /**
+             * It can happen that a withdrawal (or a full withdrawal) transactions would
+             * remove all of the liquidity from the token with a NFT token still existing.
+             * In that case the token can not be staked into the gauge, as some liquidity
+             * needs to be added to it first.
+             */
+            if (_getLiquidity() > 0) {
+                positionManager.approve(address(clGauge), tokenId);
+                clGauge.deposit(tokenId);
+            }
+        }
+    }
+
+    // slither-disable-end reentrancy-no-eth
+
+    /// @notice the constructor
+    /// @param _stratConfig the basic strategy configuration
+    /// @param _wethAddress Address of the Erc20 WETH Token contract
+    /// @param _oethbAddress Address of the Erc20 OETHb Token contract
+    /// @param _swapRouter Address of the Aerodrome Universal Swap Router
+    /// @param _nonfungiblePositionManager Address of position manager to add/remove
+    ///         the liquidity
+    /// @param _clPool Address of the Aerodrome concentrated liquidity pool
+    /// @param _clGauge Address of the Aerodrome slipstream pool gauge
+    /// @param _sugarHelper Address of the Aerodrome Sugar helper contract
+    /// @param _lowerBoundingTick Smaller bounding tick of our liquidity position
+    /// @param _upperBoundingTick Larger bounding tick of our liquidity position
+    /// @param _tickClosestToParity Tick that is closer to 1:1 price parity
+    constructor(
+        BaseStrategyConfig memory _stratConfig,
+        address _wethAddress,
+        address _oethbAddress,
+        address _swapRouter,
+        address _nonfungiblePositionManager,
+        address _clPool,
+        address _clGauge,
+        address _sugarHelper,
+        int24 _lowerBoundingTick,
+        int24 _upperBoundingTick,
+        int24 _tickClosestToParity
+    ) InitializableAbstractStrategy(_stratConfig) {
+        require(
```

Comment:
The other `require`s in this method are at the end where this is at the top. Would be nice to have all of them at a single place. I assume top would be better (but either works)

Reply:
Agreed: https://github.com/OriginProtocol/origin-dollar/pull/2184/commits/5c1f8d8cc047b206990a2eea08c97b5298b640f2